### PR TITLE
Reduce binary size in vendered build of zstd-sys if feat thin is enabled

### PR DESCRIPTION
**Pull request by NobodyXu**

### CODE REVIEW

File: zstd-safe/zstd-sys/build.rs [hunk 1]
```diff
@@ -159,9 +159,25 @@ fn compile_zstd() {
     #[cfg(feature = "thin")]
     {
         // Here we try to build a lib as thin/small as possible.
-        config.define("HUF_FORCE_DECOMPRESS_X1", Some("1"));
-        config.define("ZSTD_FORCE_DECOMPRESS_SEQUENCES_SHORT", Some("1"));
-        config.define("ZSTD_NO_INLINE ", Some("1"));
```
Comment (NobodyXu) [inline zstd-safe/zstd-sys/build.rs:164]: This is the typo which #187 intended to fix.

However, with this PR we switch to using `ZSTD_LIB_MINIFY` so this is fixed anyway.

**Review by gyscos — COMMENTED** [review]
Thanks for the work! This should indeed simplify getting a small library.

Just a few details.

File: zstd-safe/zstd-sys/build.rs [hunk 2]
```diff
@@ -159,9 +159,25 @@ fn compile_zstd() {
...
+
+        // Disable all deprecated stuff
+        config.define("ZSTD_LIB_DEPRECATED", Some("0"));
```
[the remark this answers is not in the document]
Comment (gyscos) [reply zstd-safe/zstd-sys/build.rs:164]: Actually we already set this to 0 just earlier:
https://github.com/gyscos/zstd-rs/pull/189/files#diff-d10afeb74f1c422688106d5e75cd08ac2ac0328ef71bb829a5f127f1a105e1f6L146

File: zstd-safe/zstd-sys/build.rs [hunk 3]
```diff
@@ -159,9 +159,25 @@ fn compile_zstd() {
...
+
+        // ZSTD_LIB_MINIFY implies:
+        //  - HUF_FORCE_DECOMPRESS_X1
+        //  - ZSTD_FORCE_DECOMPRESS_SEQUENCES_SHORT
+        //  - ZSTD_NO_INLINE
+        //  - ZSTD_STRIP_ERROR_STRINGS, removes the error messages that are
+        //    otherwise returned by ZSTD_getErrorName
```
[the remark this answers is not in the document]
Comment (NobodyXu) [reply zstd-safe/zstd-sys/build.rs:168]: Thanks for verifying `ZSTD_STRIP_ERROR_STRINGS` build macro is safe to use!

File: zstd-safe/zstd-sys/build.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (NobodyXu) [reply zstd-safe/zstd-sys/build.rs]: I've removed this `config.define` as deprecated stuff are already excluded.

**Comment by gyscos** [comment]
Thanks again!

**Pull request merged**